diff -up rapidfuzz-cpp-1.10.4/fuzzing/fuzzing.hpp.orig rapidfuzz-cpp-1.10.4/fuzzing/fuzzing.hpp
--- rapidfuzz-cpp-1.10.4/fuzzing/fuzzing.hpp.orig	2023-01-05 22:01:54.296770654 -0500
+++ rapidfuzz-cpp-1.10.4/fuzzing/fuzzing.hpp	2023-01-05 22:02:06.688879926 -0500
@@ -11,7 +11,7 @@ static inline bool extract_strings(const
     }
     uint32_t len1 = *(uint32_t*)data;
 
-    if (len1 + sizeof(len1) > size) {
+    if (len1 > size - sizeof(len1)) {
         return false;
     }
 
